test/shared/src/main/scala/zio/test/Assertion.scala

```diff
@@ -563,13 +678,45 @@ object Assertion extends AssertionVariants {
       case Left(_)  => None
     }
 
+  /**
+   * Makes a new assertion that requires an Either is Right.
+   */
+  val isRight: Assertion[Either[Any, Any]] =
+    isRight(anything)
+
   /**
    * Makes a new assertion that requires a Some value satisfying the specified
    * assertion.
    */
   def isSome[A](assertion: Assertion[A]): Assertion[Option[A]] =
     Assertion.assertionRec("isSome")(param(assertion))(assertion)(identity(_))
 
+  /**
+   * Makes a new assertion that requires an Option is Some.
+   */
+  val isSome: Assertion[Option[Any]] =
+    isSome(anything)
+
+  /**
+   * Makes a new assertion that requires a Seq is sorted.
+   */
+  def isSorted[A](implicit ord: Ordering[A]): Assertion[Seq[A]] = {
+    @scala.annotation.tailrec
+    def loop(seq: Seq[A]): Boolean = seq match {
```

Comment:
May want to use iterator here since some collections don't support efficient restructuring like this. That could also allow us to expand input type to `Iterable`. Some iterables like sets or maps don't have a well defined ordering but others like sorted sets and maps do, so could be nice to let users test this. What do you think?

Reply:
I have re-factored this code to use an iterator.